lib/common/schemas.c

```diff
@@ -1053,6 +1058,23 @@ apply_upgrade(const xmlNode *original_xml, int schema_index, gboolean to_logs)
         free(transform_leave);
     }
 
+    if (final == NULL) {
+        return NULL;
+    }
```

Comment:
If you move this up above the previous block, you can also get rid of the `(final != NULL)` part of it.

Reply:
final can become NULL in the previous block